commit 860b334e9a10e99a4a1191d1acfe131475888cdb
Author: Luiz Oliveira <ziuloliveira@gmail.com>
Date:   Tue Sep 1 10:16:50 2015 -0300

    apt-cli-search: Sorting package by version using
    
    Thanks: David Kalnischkies by the tips about debVS::CmpVersion
    
    Signed-off-by: Luiz Oliveira <ziuloliveira@gmail.com>

diff --git a/apt-private/private-package-info.cc b/apt-private/private-package-info.cc
index 66600e4..d0407d8 100644
--- a/apt-private/private-package-info.cc
+++ b/apt-private/private-package-info.cc
@@ -10,7 +10,7 @@
 #include <apt-pkg/depcache.h>
 #include <apt-pkg/pkgcache.h>
 #include <apt-pkg/cacheiterators.h>
-
+#include <apt-pkg/debversion.h>
 #include <apt-private/private-output.h>
 #include <apt-private/private-cachefile.h>
 #include <apt-private/private-package-info.h>
@@ -123,8 +123,9 @@ bool OrderByReverseAlphabetic (const PackageInfo &a, const PackageInfo &b)
 
 bool OrderByVersion (const PackageInfo &a, const PackageInfo &b)
 {
-   if(a.version() == b.version())
-      return a.name() < b.name();
+   int result = debVS.CmpVersion(a.version(),b.version());
+   if(result > 0)
+      return true;
    else
-      return a.version() > b.version();
+      return false;
 }
